lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
   end
 
   class BaseEventBuilder
+    CUSTOM_ATTRIBUTE_FEATURE_TYPE = 'custom'
+
     attr_reader :config
-    attr_reader :params
 
     def initialize(config)
       @config = config
-      @params = {}
     end
 
     private
 
-    def add_common_params(user_id, attributes)
+    def get_common_event_params(user_id, attributes)
       # Add params which are used in both conversion and impression events.
-      #
-      # user_id - ID for user.
-      # attributes - Hash representing user attributes and values which need to be recorded.
-
-      add_project_id
-      add_account_id
-      add_user_id(user_id)
-      add_attributes(attributes)
-      add_source
-      add_time
+      # 
+      # user_id -    +String+ ID for user
+      # attributes - +Hash+ representing user attributes and values which need to be recorded.
+      # 
+      # Returns +Hash+ Common event params
+
+      visitor_attributes = []
+
+      unless attributes.nil?
+        attributes.keys.each do |attribute_key|
+          attribute_value = attributes[attribute_key]
+          #next unless attribute_value # Omit falsy attribute values
```

Comment:
Looks like something is missing here.

Reply:
can you please point out what is missing?

Reply:
Null check on attribute value. Like here : https://github.com/msohailhussain/php-sdk/blob/pr_eventbuilderpayload/src/Optimizely/Event/Builder/EventBuilder.php#L88

Reply:
@oakbani its right there. unless attributes.nil?